src/db.rs

```diff
@@ -1124,18 +1124,23 @@ impl<T: ThreadMode, D: DBInner> DBCommon<T, D> {
         K: AsRef<[u8]>,
         I: IntoIterator<Item = K>,
     {
-        let (keys, keys_sizes): (Vec<Box<[u8]>>, Vec<_>) = keys
+        let keys_data: Vec<_> = keys
             .into_iter()
             .map(|k| {
                 let k = k.as_ref();
                 (Box::from(k), k.len())
             })
-            .unzip();
-        let ptr_keys: Vec<_> = keys.iter().map(|k| k.as_ptr() as *const c_char).collect();
+            .collect();
+        
+        let ptr_keys: Vec<_> = keys_data
+            .iter()
+            .map(|(k, _)| k.as_ptr() as *const c_char)
+            .collect();
+        let keys_sizes: Vec<_> = keys_data.iter().map(|(_, size)| *size).collect();
```

Comment:
> Eliminated unnecessary collection creation by avoiding unzip()

I don't get this part: instead of using unzip you are just using another collect. What is the difference?